src/formatDate.js

```diff
@@ -50,7 +50,54 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromDate = date.split(fromFormat[3]);
+  const dates = {};
+
+  if (fromFormat[0] === 'YY'
+  && (toFormat[0] === 'YYYY' || toFormat[2] === 'YYYY')) {
+    if (fromDate[0].toString() < 30) {
+      fromDate[0] = '20' + fromDate[0];
+    } else {
+      fromDate[0] = '19' + fromDate[0];
+    }
+    fromFormat[0] = 'YYYY';
+  }
+
+  if (fromFormat[2] === 'YY'
+  && (toFormat[0] === 'YYYY' || toFormat[2] === 'YYYY')) {
+    if (fromDate[0].toString() > 31) {
```

Comment:
If you have some similar checks, you can create separate variable (for example for (toFormat[0] === 'YYYY' || toFormat[2] === 'YYYY') and use it in your code